docs/dev.md

```diff
@@ -0,0 +1,420 @@
+}
+```
+
+And the content of `/etc/mesos/overlay/master-agent-config.json` is:
+
+```json
+{
+    "cni_dir": "/etc/mesos/cni",
+    "cni_data_dir": "/var/run/mesos/cni/networks",
+    "network_config" : {
+        "allocate_subnet": true,
+        "mesos_bridge": false,
+        "docker_bridge": false,
+        "overlay_mtu": 1420,
+        "enable_ipv6": false
+    },
+    "max_configuration_attempts": 5
+}
+```
+
+Now let's start Mesos agent on `node-2`:
+
+```sh
+mesos-agent \
+    --containerizers=mesos \
+    --hostname=$NODE_2_IP \
+    --image_providers=docker \
+    --ip=0.0.0.0 \
+    --isolation=docker/runtime,filesystem/linux,volume/sandbox_path \
+    --log_dir=/var/log/mesos \
+    --master=zk://$NODE_1_IP:2181/mesos \
+    --port=5051 \
+    --work_dir=/var/lib/mesos/agent \
+    --no-systemd_enable_support \
+    --modules=file:///etc/mesos/agent-modules.json
+```
+
+where the content of `/etc/mesos/agent-modules.json` is:
+
+```json
+{
+    "libraries": [{
+        "file": "/usr/lib/mesos/libmesos_network_overlay.so",
+        "modules": [{
+            "name": "com_mesosphere_mesos_OverlayAgentManager",
+            "parameters": [{
+                "key": "agent_config",
+                "value": "/etc/mesos/overlay/agent-config.json"
+            }]
+        }]
+    }]
+}
+```
+
+And the content of `/etc/mesos/overlay/agent-config.json` is:
+
+```json
+{
+    "master": "$NODE_1_IP:5050",
+    "cni_dir":"/etc/mesos/cni",
+    "cni_data_dir": "/var/run/mesos/cni/networks",
+    "network_config": {
+        "allocate_subnet": true,
+        "mesos_bridge": true,
+        "docker_bridge": false,
+        "overlay_mtu": 1420,
+        "enable_ipv6": false
+    },
+    "max_configuration_attempts": 5
+}
+```
+
+`dcos-net` depends on Mesos-DNS, so let's tilt it up too. First, build it
+following the following [instructions](https://github.com/mesosphere/mesos-dns),
+and then run on `node-1`:
+
+```sh
+mesos-dns -config /etc/mesos-dns.json
+```
+
+where the content of `/etc/mesos-dns.json` is:
+
+```json
+{
+    "zk": "zk://$NODE_1_IP:2181/mesos",
+    "refreshSeconds": 30,
+    "ttl": 60,
+    "domain": "mesos",
+    "port": 61053,
+    "resolvers": ["8.8.8.8"],
+    "timeout": 5,
+    "listener": "0.0.0.0",
+    "email": "root.mesos-dns.mesos",
+    "IPSources": ["host", "netinfo"],
+    "SetTruncateBit": true
+}
+```
+
+`dcos-net` expects some network interfaces to be up and configured. Let's do
+this on both nodes:
+
+```sh
+modprobe ip_vs_wlc
+modprobe dummy
+
+iptables --wait -A FORWARD -j ACCEPT
+iptables --wait -t nat -I POSTROUTING -m ipvs --ipvs --vdir ORIGINAL \
+         --vmethod MASQ -m comment \
+         --comment Minuteman-IPVS-IPTables-masquerade-rule \
+         -j MASQUERADE
+ip link add minuteman type dummy
+ip link set minuteman up
+ip link add spartan type dummy
+ip link set spartan up
+ip addr add 198.51.100.1/32 dev spartan
+ip addr add 198.51.100.2/32 dev spartan
+ip addr add 198.51.100.3/32 dev spartan
```

Comment:
I had trouble with that on a mac. I believe that this is due to trying to modprobe on a docker container on docker on mac. Somehow if you use dcos-e2e gets around that networking on a different way. Is it worth mentioning here how that can be done, or at least just mentioned that if you are on a mac with docker, you are out of luck?

Reply:
@Fabs These commands should work fine if you run them in docker with `--privileged`.

Reply:
Hum, I will research a little more. I tried with `--privileged` before, but although I got some extra capabilities, `modprobe`, `rmmod` and etc was not available. Maybe I am missing something here. I will look more carefully. Thanks.